### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by AleksandrPok**

### CODE REVIEW

File: src/main/java/cinema/controller/InjectData.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    @PostConstruct
+    private void inject() {
+        Role userRole = new Role();
+        userRole.setRoleName(Roles.USER);
+        roleService.add(userRole);
+        Role adminRole = new Role();
+        adminRole.setRoleName(Roles.ADMIN);
+        roleService.add(adminRole);
+        User admin = new User();
+        admin.setRoles(List.of(adminRole));
```
Comment (Max-Pochepets) [inline src/main/java/cinema/controller/InjectData.java:38]: Your admin will have shopping cart. Why is he not USER?

File: src/main/java/cinema/service/mappers/UserMapper.java [hunk 2]
```diff
@@ -2,10 +2,19 @@
...
 
 @Component
 public class UserMapper {
+    private final RoleService roleService;
+
+    @Autowired
+    public UserMapper(RoleService roleService) {
+        this.roleService = roleService;
+    }
+
```
Comment (Max-Pochepets) [inline src/main/java/cinema/service/mappers/UserMapper.java]: Why?

**Review by Max-Pochepets — APPROVED** [review]

File: src/main/java/cinema/config/SecurityConfig.java [hunk 3]
```diff
@@ -1,37 +1,40 @@
...
-                .withUser("testUser@test.com")
-                .password(getEncoder().encode("testPassword"))
-                .roles("USER");
+        managerBuilder.userDetailsService(userDetailsService).passwordEncoder(passwordEncoder);
     }
 
     protected void configure(HttpSecurity httpSecurity) throws Exception {
         httpSecurity.authorizeRequests()
-                .anyRequest().authenticated()
+                .antMatchers("/register").permitAll()
```
Comment (RomanDubovskyi) [inline src/main/java/cinema/config/SecurityConfig.java]: pay attention to the message Sofia sent in general channel

File: src/main/java/cinema/dao/UserDaoImpl.java [hunk 4]
```diff
@@ -41,8 +41,9 @@ public User add(User user) {
     @Override
     public Optional<User> findByEmail(String email) {
         try (Session session = sessionFactory.openSession()) {
-            return session.createQuery("FROM User WHERE email = :email", User.class)
-                   .setParameter("email", email).uniqueResultOptional();
+            return session.createQuery("SELECT u FROM User u "
+                    + "LEFT JOIN FETCH u.roles WHERE u.email = :email", User.class)
```
Comment (RomanDubovskyi) [inline src/main/java/cinema/dao/UserDaoImpl.java]: let's use inner join, there shouldn't be situation when user doesn't have role

File: src/main/java/cinema/dao/UserDaoImpl.java [hunk 5]
```diff
@@ -51,7 +52,9 @@ public User add(User user) {
     @Override
     public Optional<User> get(Long id) {
         try (Session session = sessionFactory.openSession()) {
-            return Optional.ofNullable(session.get(User.class, id));
+            return session.createQuery("SELECT u FROM User u "
+                    + "LEFT JOIN FETCH u.roles WHERE u.id = :id", User.class)
```
Comment (RomanDubovskyi) [inline src/main/java/cinema/dao/UserDaoImpl.java]: same here

File: src/main/java/cinema/model/Roles.java [hunk 6]
```diff
@@ -0,0 +1,6 @@
+package cinema.model;
+
+public enum Roles {
```
Comment (RomanDubovskyi) [inline src/main/java/cinema/model/Roles.java]:
```suggestion
public enum RoleName {
```

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**